### PR TITLE
Bug 1109649 - Display page loading errors

### PR DESCRIPTION
**Pull request by lyle-luan**
A simply rough implementation.
Follow the hints in comment 3.
Looking forward for some advices.

### CODE REVIEW

**Comment by lyle-luan** [comment]
Please ignore this, I made a big mistake here.
I should load HTML to display error, and I am working on it.
Also I will test a bit manually for some common scenarios, thanks for Stefan Arentz's suggestion.

**Comment by st3fan** [comment]
Just a quick note that I'll take a look Monday to see if we can merge this

**Comment by lyle-luan** [comment]
Load web page to display errors, make it look like the way Safari deals with errors.

**Comment by st3fan** [comment]
I'll look at this bug tonight.

File: ClientTests/WebServerTests.swift [hunk 1]
```diff
@@ -16,15 +16,16 @@ class WebServerTests: XCTestCase {
...
     }
 
     override func setUp() {
         super.setUp()
         setupWebServer()
+
+        if webServer.startWithPort(0, bonjourName: nil) == false {
+            XCTFail("Can't start the GCDWebServer")
+        }
+        webServerBase = "http://localhost:\(webServer.port)"
```
Comment (st3fan) [inline ClientTests/WebServerTests.swift]: Why did you move this out of `setupWebServer()` ?

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -183,6 +192,31 @@ extension BrowserViewController: WKNavigationDelegate {
...
+        
+        resultString += "code: \(error.code)\n"
+        resultString += "domain: \(error.domain)\n"
+        resultString += "userInfo: \(error.userInfo)\n"
+        resultString += "localizedDescription: \(error.localizedDescription)\n"
+        resultString += "localizedRecoveryOptions: \(error.localizedRecoveryOptions)\n"
+        resultString += "localizedRecoverySuggestion: \(error.localizedRecoverySuggestion)\n"
+        resultString += "localizedFailureReason: \(error.localizedFailureReason)\n"
+        resultString += "recoveryAttempter: \(error.recoveryAttempter)\n"
+        resultString += "helpAnchor: \(error.helpAnchor)\n"
```
Comment (st3fan) [inline Client/Frontend/Browser/BrowserViewController.swift]: I don't see `resultString` being used. Is this just for debugging or do you intent do include it on the page?

I think it would be interesting to display all those values on the page: that will give us a good idea of what is actually relevant and then we can decide what we want to show in a more final version.

Comment (lyle-luan) [inline Client/Frontend/Browser/BrowserViewController.swift]: Yes, that's what I want to, display this all on the error page, I forget to add a //comment.

File: ClientTests/WebServerTests.swift (hunk 1 above)
Comment (lyle-luan) [inline ClientTests/WebServerTests.swift]: Thank you for your review, I will change the func name. :]

**Comment by st3fan** [comment]
I am going to merge this with the comment that this is not perfect but it will give us a much better idea of what kind of usable information WebKit errors contain. I am also going to file a followup but to turn these error messages into a more final design.

**Comment by st3fan** [comment]
Just a quick note that on second thought we think it is better to revert this merge. There are a couple of things in the changes that we want to do in a different way and we need to give this some more thought.

**Comment by lyle-luan** [comment]
I am going to merge the tests of GCDWebServer to WebServerTests.swift, Should I working on it?
